### DISCUSSION TITLE
configure: error: --with-openssl was given but OpenSSL could not be detected  curl 8.9.1 with openssl 3.0.13 on aix with gcc compiler

### DISCUSSION THREAD

**Discussion by srini7015** [curl/curl#14856] [General]
trying to compile curl 8.9.1 with openssl 3.0.13

i have installed my openssl in my custom location. when i am tring to configure curl with custom openssl install location, i am getting the error "configure: error: --with-openssl was given but OpenSSL could not be detected".

Can someone help.

./configure --disable-shared --without-zlib --with-openssl => no issue and it is taking openssl 1.0.2t which is installed in usr/local/lib 

i installed openssl 3.0.13 in the "/home/xyz1/xyz2/jenkinsopenssl"

./configure --disable-shared --without-zlib --with-openssl=/home/xyz1/xyz2/jenkinsopenssl ==> i am getting error as below.

configure: error: --with-openssl was given but OpenSSL could not be detected

**Comment by bagder** [comment]
Typically the answer is in your `config.log`.

**Comment by srini7015** [comment]
[config.log](https://github.com/user-attachments/files/16963829/config.log)
i have attached the config.log , can you please review

**Comment by jzakrzewski** [comment]
Is it possible, that your OpenSSL has been built with `OPENSSL_NO_DEPRECATED` set?

**Reply by srini7015** [reply]
I used below command to  configure openssl before doing make for openssl
 
./Configure -DOPENSSL_TLS_SECURITY_LEVEL=0 enable-weak-ssl-ciphers enable-des threads no-shared no-engine no-hw --prefix=/home/tn044716/daripelli/opensslinstall --openssldir=/home/tn044716/daripelli/opensslinstall aix64-gcc

**Reply by jingxiangchung** [reply]
I had the same issue yesterday, and I happened to come accross your question. It took me quite some time to only realised that when compiling openssl, the openssl libraries was placed in the "lib64" folder of my openssl dir instead of the "lib" folder. Since curl configure is looking for the openssl libraries in the "lib" folder, I got the same "configure: error: --with-openssl was given but OpenSSL could not be detected"

A few solutions:
1. Soft link the required libraries (libcrypto.so and libssl.so) located in "lib64" to "lib" folder
2. When compiling the openssl, request the compiler to place the libraries in "lib"
3. During configure stage of curl, tell the configure to look at "lib64" folder using LDFLAGS

I opted the number 3 option, thus

INSTALL_DIR=/opt/gfort_compiled

#Compiling openssl:
./Configure --prefix=${INSTALL_DIR} --openssldir=${INSTALL_DIR} --with-zlib-include=${INSTALL_DIR} --with-zlib-lib=${INSTALL_DIR}
make
make install

#Compiling curl
./configure --prefix=${INSTALL_DIR} --with-zlib=${INSTALL_DIR} --with-openssl=${INSTALL_DIR} CPPFLAGS="${CPPFLAGS} -I${INSTALL_DIR}/include" **LDFLAGS="$LDFLAGS -L${INSTALL_DIR}/lib64"** CFLAGS=-fPIC
make
make check
make install

Depending on your computer, you might need to compille lipsl too before compiling curl to makes the curl configure works.

Hope this helps!

**Reply by srini7015** [reply]
I have tried this steps on my aix 7.1 machine , but still i am getting error "OpenSSL could not be detected" while doing cofigure step of curl.

**Comment by xiaoming6868** [comment]
Hello, I also encountered this problem today, but I used it on a cross-compiled development board to aarch64, and finally I tried to specify the zlib path to solve this error, you can refer to my instructions

./configure --host=aarch64-linux-gnu --build=x86_64-linux-gnu \
--prefix=/home/xiaoming/tool/curl/curlTool \
--with-ssl=/home/xiaoming/tool/openssl/opensslTool \
--with-zlib=/home/xiaoming/tool/zlib/zlibTool \
--with-ssl-lib=/home/xiaoming/tool/openssl/opensslTool/lib \
--with-ssl-include=/home/xiaoming/tool/openssl/opensslTool/include \
--with-zlib-lib=/home/xiaoming/tool/zlib/zlibTool/lib \
--with-zlib-include=/home/xiaoming/tool/zlib/zlibTool/include \
--disable-shared --enable-static

**Comment by dfandrich** [comment]
It's usually easiest and simpler setting `PKG_CONFIG_PATH` so dependent libraries can be found using pkg-config. See [INSTALL.md](https://curl.se/docs/install.html).
